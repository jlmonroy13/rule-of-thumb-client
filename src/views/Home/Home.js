import React, { useState } from 'react';
import { useStorageState } from 'react-storage-hooks';
import { equals } from 'ramda';

import RulingBox from 'components/RulingBox';
import ProgressBar from 'components/ProgressBar';
import InfoBox from 'components/InfoBox';
import RulingCard from 'components/RulingCard';
import Layout from 'components/Layout';

import data from 'constants/data.json';

const Home = () => {
  const [showInfoBox, setShowInfoBox] = useState(true);
  const [rulings, setRulings] = useStorageState(
    localStorage,
    'rulings',
    data.rulings,
  );

  const onCloseInfoBox = () => setShowInfoBox(false);

  const updateRulingsArray = ({ id, ...voteData }) => (ruling) => {
    if (equals(id, ruling.id)) {
      return {
        ...ruling,
        ...voteData,
      };
    }

    return ruling;
  };

  const onSetVote = (voteData) => {
    const updatedRulings = rulings.map(updateRulingsArray(voteData));
    setRulings(updatedRulings);
  };

  const onRenderRulings = (ruling) => (
    <div key={ruling.id} className="col-lg-6">
      <RulingCard {...ruling} onSetVote={onSetVote} />
    </div>
  );

  return (
    <Layout>
      <section className="home__main">
        <h1 className="d-none">Last Ruling</h1>
        <div className="container">
          <RulingBox
            title="Pope Francis?"
            description="Heâ€™s talking tough on clergy sexual abuse, but is he just another papal pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up)"
            wikiUrl="https://en.wikipedia.org/wiki/Catholic_Church_sexual_abuse_cases"
          />
        </div>
        <div className="home__main-footer">
          <ProgressBar label="closing in" value={22} valueLabel="days" />
        </div>
      </section>
      <section className="home__rulings">
        <div className="container">
          {showInfoBox && (
            <InfoBox
              title="Be counted"
              subtitle="Speak out. Be heard."
              message="Rule of Thumb is a crowd sourced court of public opinion where anyone and everyone can speak out and speak freely. It&lsquo;s easy: You share your opinion, we analyze and put the data in a public report."
              onClose={onCloseInfoBox}
            />
          )}
          <h1 className="home__rulings-title mb-7">Previous Rulings</h1>
          <div className="row">{rulings.map(onRenderRulings)}</div>
          <div className="home__advise">
            <div className="home__advise-mask" />
            <p className="home__advise-text">
              Is there anyone else you would want us to add?
            </p>
            <button className="home__advise-btn">Submit a Name</button>
          </div>
        </div>
      </section>
    </Layout>
  );
};

export default Home;
