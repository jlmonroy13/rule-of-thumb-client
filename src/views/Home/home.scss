.home {
  &__main {
    height: 100vh;
    background-image: url('../../assets/images/pope.jpg');
    background-size: cover;
    position: relative;
    padding-top: 86px;
    display: flex;
    align-items: center;

    &-footer {
      position: absolute;
      bottom: 0;
      left: 0;
      width: 100%;
      display: flex;
    }

    &-footer-left {
      width: 35%;
      background-color: rgba($black, 0.3);
      display: flex;
      justify-content: flex-end;
      padding-right: 20px;
      align-items: center;
      position: relative;

      &:after {
        position: absolute;
        right: -12px;
        content: '';
        width: 0;
        height: 0;
        border-style: solid;
        border-width: 8px 0 8px 12px;
        border-color: transparent transparent transparent rgba($black, 0.3);
      }
    }

    &-footer-right {
      flex-grow: 1;
      background-color: rgba($white, 0.7);
      @include font-size(37);
      line-height: 50px;
      font-weight: 300;
      padding: 0 20px;
    }
  }
}
