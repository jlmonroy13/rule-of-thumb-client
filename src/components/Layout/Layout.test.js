/**
 *
 * Tests for Layout
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import Home from 'views/Home';

import Layout from './index';

describe('<Layout />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Router>
        <Layout>
          <Home />
        </Layout>
      </Router>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Router>
        <Layout>
          <Home />
        </Layout>
      </Router>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
