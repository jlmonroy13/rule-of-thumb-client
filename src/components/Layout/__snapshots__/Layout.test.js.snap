// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Should render and match the snapshot 1`] = `
<div
  class="relative"
>
  <header
    class="header"
  >
    <div
      class="header__mask"
    />
    <div
      class="container"
    >
      <div
        class="d-flex justify-content-between align-items-center"
      >
        <a
          aria-current="page"
          class="header__title active"
          href="/"
        >
          <h1>
            Rule of Thumb.
          </h1>
        </a>
        <nav
          class="header__nav "
        >
          <button
            class="d-block d-md-none mb-4"
          >
            <svg
              class="icon icon-arrow-right "
              data-testid="icon-component"
            >
              <use
                xlink:href="#icon-arrow-right"
              />
            </svg>
          </button>
          <a
            class="header__nav-item"
            href="/past-trials"
          >
            Past Trials
          </a>
          <a
            class="header__nav-item"
            href="/how-it-works"
          >
            How It Works
          </a>
          <a
            class="header__nav-item"
            href="/login"
          >
            Log In/Sign Up
          </a>
          <button>
            <svg
              class="icon icon-search header__icon"
              data-testid="icon-component"
            >
              <use
                xlink:href="#icon-search"
              />
            </svg>
          </button>
        </nav>
        <button
          class="d-block d-md-none"
        >
          <div
            class="header__hamburger"
          >
            <span
              class="header__hamburger-line"
            />
            <span
              class="header__hamburger-line"
            />
            <span
              class="header__hamburger-line"
            />
          </div>
        </button>
      </div>
    </div>
  </header>
  <div
    class="relative"
  >
    <header
      class="header"
    >
      <div
        class="header__mask"
      />
      <div
        class="container"
      >
        <div
          class="d-flex justify-content-between align-items-center"
        >
          <a
            aria-current="page"
            class="header__title active"
            href="/"
          >
            <h1>
              Rule of Thumb.
            </h1>
          </a>
          <nav
            class="header__nav "
          >
            <button
              class="d-block d-md-none mb-4"
            >
              <svg
                class="icon icon-arrow-right "
                data-testid="icon-component"
              >
                <use
                  xlink:href="#icon-arrow-right"
                />
              </svg>
            </button>
            <a
              class="header__nav-item"
              href="/past-trials"
            >
              Past Trials
            </a>
            <a
              class="header__nav-item"
              href="/how-it-works"
            >
              How It Works
            </a>
            <a
              class="header__nav-item"
              href="/login"
            >
              Log In/Sign Up
            </a>
            <button>
              <svg
                class="icon icon-search header__icon"
                data-testid="icon-component"
              >
                <use
                  xlink:href="#icon-search"
                />
              </svg>
            </button>
          </nav>
          <button
            class="d-block d-md-none"
          >
            <div
              class="header__hamburger"
            >
              <span
                class="header__hamburger-line"
              />
              <span
                class="header__hamburger-line"
              />
              <span
                class="header__hamburger-line"
              />
            </div>
          </button>
        </div>
      </div>
    </header>
    <section
      class="home__main"
    >
      <h1
        class="d-none"
      >
        Last Ruling
      </h1>
      <div
        class="container"
      >
        <div
          class="ruling-box"
        >
          <span
            class="text-small"
          >
            What's your opinion on
          </span>
          <h1
            class="ruling-box__title"
          >
            Pope Francis?
          </h1>
          <p
            class="ruling-box__description"
          >
            He’s talking tough on clergy sexual abuse, but is he just another papal pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up)
          </p>
          <div
            class="mb-6"
          >
            <svg
              class="icon icon-wikipedia mr-1"
              data-testid="icon-component"
            >
              <use
                xlink:href="#icon-wikipedia"
              />
            </svg>
            <a
              class="ruling-box__link"
              href="https://en.wikipedia.org/wiki/Catholic_Church_sexual_abuse_cases"
              rel="noopener noreferrer"
              target="_blank"
            >
              More information
            </a>
          </div>
          <span
            class="font-weight-bold text-larger"
          >
            What's Your Verdict?
          </span>
          <div
            class="ruling-box__buttons"
          >
            <button
              class="ruling-box__button"
            >
              <svg
                class="icon icon-thumb ruling-box__icon ruling-box__icon--up"
                data-testid="icon-component"
              >
                <use
                  xlink:href="#icon-thumb"
                />
              </svg>
            </button>
            <button
              class="ruling-box__button ruling-box__button--yellow"
            >
              <svg
                class="icon icon-thumb ruling-box__icon"
                data-testid="icon-component"
              >
                <use
                  xlink:href="#icon-thumb"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="home__main-footer"
      >
        <div
          class="d-flex"
        >
          <div
            class="progress-bar__lapsed"
          >
            <span
              class="text-white text-uppercase text-small"
            >
              closing in
            </span>
          </div>
          <div
            class="progress-bar__remaining"
          >
            <b
              class="mr-2"
            >
              22
            </b>
            <span>
              days
            </span>
          </div>
        </div>
      </div>
    </section>
    <section
      class="home__rulings"
    >
      <div
        class="container"
      >
        <div
          class="info-box"
          data-testid="info-box"
        >
          <div
            class="info-box__container"
          >
            <div
              class="mr-5"
            >
              <span
                class="info-box__subtitle"
              >
                Speak out. Be heard.
              </span>
              <span
                class="info-box__title"
              >
                Be counted
              </span>
            </div>
            <p
              class="info-box__message"
            >
              Rule of Thumb is a crowd sourced court of public opinion where anyone and everyone can speak out and speak freely. It‘s easy: You share your opinion, we analyze and put the data in a public report.
            </p>
          </div>
          <button
            class="info-box__button"
            data-testid="info-box-close-button"
          >
            ✕
          </button>
        </div>
        <h1
          class="home__rulings-title mb-7"
        >
          Previous Rulings
        </h1>
        <div
          class="row"
        >
          <div
            class="col-lg-6"
          >
            <div
              class="ruling-card"
              style="background-image: url(west.jpg);"
            >
              <div
                class="ruling-card__mask"
              />
              <div
                class="ruling-card__body"
              >
                <div
                  class="position-relative"
                >
                  <span
                    class="ruling-card__icon"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </span>
                  <h2
                    class="ruling-card__title"
                  >
                    Kanye West
                  </h2>
                </div>
                <span
                  class="text-small mb-3 d-block"
                >
                  <b>
                    2 months
                     ago
                  </b>
                   in
                   
                  Entertainment
                </span>
                <p
                  class="ruling-card__description"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.
                </p>
                <div
                  class="ruling-card__buttons-container"
                >
                  <button
                    class="ruling-card__thumb-btn mr-2 "
                    data-testid="thumb-up-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__thumb-btn mr-3 "
                    data-testid="thumb-down-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__button"
                    data-testid="vote-button"
                  >
                    Vote 
                    now
                  </button>
                </div>
              </div>
              <div
                class="ruling-card__footer"
              >
                <div
                  class="ruling-card__footer-percentage ruling-card__footer-percentage--up"
                  style="width: 0%;"
                >
                  <svg
                    class="icon icon-thumb ruling-card__footer-icon"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                  <span>
                    0
                    %
                  </span>
                </div>
                <div
                  class="ruling-card__footer-percentage"
                  style="width: 0%;"
                >
                  <span>
                    0
                    %
                  </span>
                  <svg
                    class="icon icon-thumb ml-2"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-lg-6"
          >
            <div
              class="ruling-card"
              style="background-image: url(zuckerberg.jpg);"
            >
              <div
                class="ruling-card__mask"
              />
              <div
                class="ruling-card__body"
              >
                <div
                  class="position-relative"
                >
                  <span
                    class="ruling-card__icon"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </span>
                  <h2
                    class="ruling-card__title"
                  >
                    Mark Zuckerberg
                  </h2>
                </div>
                <span
                  class="text-small mb-3 d-block"
                >
                  <b>
                    3 months
                     ago
                  </b>
                   in
                   
                  Business
                </span>
                <p
                  class="ruling-card__description"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.
                </p>
                <div
                  class="ruling-card__buttons-container"
                >
                  <button
                    class="ruling-card__thumb-btn mr-2 "
                    data-testid="thumb-up-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__thumb-btn mr-3 "
                    data-testid="thumb-down-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__button"
                    data-testid="vote-button"
                  >
                    Vote 
                    now
                  </button>
                </div>
              </div>
              <div
                class="ruling-card__footer"
              >
                <div
                  class="ruling-card__footer-percentage ruling-card__footer-percentage--up"
                  style="width: 60%;"
                >
                  <svg
                    class="icon icon-thumb ruling-card__footer-icon"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                  <span>
                    60
                    %
                  </span>
                </div>
                <div
                  class="ruling-card__footer-percentage"
                  style="width: 40%;"
                >
                  <span>
                    40
                    %
                  </span>
                  <svg
                    class="icon icon-thumb ml-2"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-lg-6"
          >
            <div
              class="ruling-card"
              style="background-image: url(kirchner.jpg);"
            >
              <div
                class="ruling-card__mask"
              />
              <div
                class="ruling-card__body"
              >
                <div
                  class="position-relative"
                >
                  <span
                    class="ruling-card__icon"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </span>
                  <h2
                    class="ruling-card__title"
                  >
                    Cristina Fernández de Kirchner
                  </h2>
                </div>
                <span
                  class="text-small mb-3 d-block"
                >
                  <b>
                    3 months
                     ago
                  </b>
                   in
                   
                  Politics
                </span>
                <p
                  class="ruling-card__description"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.
                </p>
                <div
                  class="ruling-card__buttons-container"
                >
                  <button
                    class="ruling-card__thumb-btn mr-2 "
                    data-testid="thumb-up-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__thumb-btn mr-3 "
                    data-testid="thumb-down-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__button"
                    data-testid="vote-button"
                  >
                    Vote 
                    now
                  </button>
                </div>
              </div>
              <div
                class="ruling-card__footer"
              >
                <div
                  class="ruling-card__footer-percentage ruling-card__footer-percentage--up"
                  style="width: 50%;"
                >
                  <svg
                    class="icon icon-thumb ruling-card__footer-icon"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                  <span>
                    50
                    %
                  </span>
                </div>
                <div
                  class="ruling-card__footer-percentage"
                  style="width: 50%;"
                >
                  <span>
                    50
                    %
                  </span>
                  <svg
                    class="icon icon-thumb ml-2"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-lg-6"
          >
            <div
              class="ruling-card"
              style="background-image: url(yousafzai.png);"
            >
              <div
                class="ruling-card__mask"
              />
              <div
                class="ruling-card__body"
              >
                <div
                  class="position-relative"
                >
                  <span
                    class="ruling-card__icon"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb "
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </span>
                  <h2
                    class="ruling-card__title"
                  >
                    Malala Yousafzai
                  </h2>
                </div>
                <span
                  class="text-small mb-3 d-block"
                >
                  <b>
                    4 months
                     ago
                  </b>
                   in
                   
                  Entertainment
                </span>
                <p
                  class="ruling-card__description"
                >
                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Consectetur, voluptates magnam excepturi libero ipsum ipsa eligendi voluptate praesentium.
                </p>
                <div
                  class="ruling-card__buttons-container"
                >
                  <button
                    class="ruling-card__thumb-btn mr-2 "
                    data-testid="thumb-up-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb ruling-card__thumb--up"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__thumb-btn mr-3 "
                    data-testid="thumb-down-button"
                  >
                    <svg
                      class="icon icon-thumb ruling-card__thumb"
                      data-testid="icon-component"
                    >
                      <use
                        xlink:href="#icon-thumb"
                      />
                    </svg>
                  </button>
                  <button
                    class="ruling-card__button"
                    data-testid="vote-button"
                  >
                    Vote 
                    now
                  </button>
                </div>
              </div>
              <div
                class="ruling-card__footer"
              >
                <div
                  class="ruling-card__footer-percentage ruling-card__footer-percentage--up"
                  style="width: 25%;"
                >
                  <svg
                    class="icon icon-thumb ruling-card__footer-icon"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                  <span>
                    25
                    %
                  </span>
                </div>
                <div
                  class="ruling-card__footer-percentage"
                  style="width: 75%;"
                >
                  <span>
                    75
                    %
                  </span>
                  <svg
                    class="icon icon-thumb ml-2"
                    data-testid="icon-component"
                  >
                    <use
                      xlink:href="#icon-thumb"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="home__advise"
        >
          <div
            class="home__advise-mask"
          />
          <p
            class="home__advise-text"
          >
            Is there anyone else you would want us to add?
          </p>
          <button
            class="home__advise-btn"
          >
            Submit a Name
          </button>
        </div>
      </div>
    </section>
    <div
      class="container"
    >
      <div
        class="footer"
      >
        <nav
          class="footer__nav"
        >
          <a
            class="footer__nav-item"
            href="/terms-and-conditions"
          >
            Terms and Conditions
          </a>
          <a
            class="footer__nav-item"
            href="/privacy-policy"
          >
            Privacy Policy
          </a>
          <a
            class="footer__nav-item"
            href="/contact-us"
          >
            Contact us
          </a>
        </nav>
        <div
          class="footer__social-networks"
        >
          <span
            class="footer__text"
          >
            Follow Us
          </span>
          <a
            href="https://www.facebook.com/pages/category/Community/FB-Rules-of-Thumb-318882558219309/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon icon-facebook footer__icon footer__icon--facebook"
              data-testid="icon-component"
            >
              <use
                xlink:href="#icon-facebook"
              />
            </svg>
          </a>
          <a
            href="https://twitter.com/ruleofthumbbiz"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="icon icon-twitter footer__icon footer__icon--twitter"
              data-testid="icon-component"
            >
              <use
                xlink:href="#icon-twitter"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="footer"
    >
      <nav
        class="footer__nav"
      >
        <a
          class="footer__nav-item"
          href="/terms-and-conditions"
        >
          Terms and Conditions
        </a>
        <a
          class="footer__nav-item"
          href="/privacy-policy"
        >
          Privacy Policy
        </a>
        <a
          class="footer__nav-item"
          href="/contact-us"
        >
          Contact us
        </a>
      </nav>
      <div
        class="footer__social-networks"
      >
        <span
          class="footer__text"
        >
          Follow Us
        </span>
        <a
          href="https://www.facebook.com/pages/category/Community/FB-Rules-of-Thumb-318882558219309/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="icon icon-facebook footer__icon footer__icon--facebook"
            data-testid="icon-component"
          >
            <use
              xlink:href="#icon-facebook"
            />
          </svg>
        </a>
        <a
          href="https://twitter.com/ruleofthumbbiz"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="icon icon-twitter footer__icon footer__icon--twitter"
            data-testid="icon-component"
          >
            <use
              xlink:href="#icon-twitter"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
